### ISSUE TITLE
Add plugin profiling for ProcIntercept

### ISSUE THREAD

**Issue by XertroV** [openplanet-nl/openplanet-issues#180]
Currently, I do not think that the total execution time of AngelScript code executed via `InterceptProc` are factored into the plugin profiling stats (which makes sense). This means that I can't measure the performance of that code which is not ideal since millisecond resolution isn't enough when measuring multiple sub-millisecond functions.

One use case I'm considering for this is a safety feature with MLHook: if more than x time has been taken during a single game frame, abort ASAP.

I had a lot of crashes y/d and I strongly suspect it is b/c some code took too long via `InterceptProc`.

**Comment by codecat** [comment]
Why do you think it's crashing because of intercepted procs taking too long?

**Comment by XertroV** [comment]
It's only a suspicion. I think it because I wrote up a data structure that, on benchmarking, took 100s of ms to insert a few thousand object references, but: i could use the data structure in other places to benchmark it and i was ingesting 100s of events sometimes. My hypothesis is introducing milliseconds of latency at various points during execution is really bad.
I struggled to get logs out, it seemed like things crashed before OP had a chance to write much out. Sometimes a partial line was written.

I had suspicions about other possible issues, but `try{}catch{}` didn't help and i could get error messages out sometimes when I did stuff I knew was wrong. The crashes that were problematic were all hard-crashes tho. no popup error, just freeze and crash to desktop.

**Comment by codecat** [comment]
You could test this suspicion by calling [`Dev::Sleep`](https://openplanet.dev/docs/api/Dev/Sleep) inside of your proc intercept. Though I would think it doesn't crash because of a time constraint, but rather something you are doing inside of the hook.

**Comment by XertroV** [comment]
I'll test it out. You might be right.

**Comment by XertroV** [comment]
I used 2 methods for slowing stuff down: dev::sleep and looping through 1000 to 10_000_000 addition operations.

Neither crashed the game, so seems you were right. Possibly it's not even to do with doing something in the hook itself.

I did confirm that OP doesn't count execution time via intercepts, though. I didn't get any laggy plugin warnings despite some of the time-wasting executions lasting more than 2s.
